ingestion/src/metadata/profiler/metrics/hybrid/cardinality_distribution.py

```diff
@@ -0,0 +1,240 @@
+#  Copyright 2025 Collate
+#  Licensed under the Collate Community License, Version 1.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#  https://github.com/open-metadata/OpenMetadata/blob/main/ingestion/LICENSE
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+"""
+Cardinality Distribution Metric definition
+"""
+from typing import Any, Dict, Optional
+
+from sqlalchemy import case, column, func
+from sqlalchemy.orm import DeclarativeMeta, Session
+
+from metadata.profiler.metrics.core import HybridMetric
+from metadata.profiler.metrics.static.count import Count
+from metadata.profiler.metrics.static.distinct_count import DistinctCount
+from metadata.profiler.orm.registry import is_concatenable, is_enum
+from metadata.utils.logger import profiler_logger
+
+logger = profiler_logger()
+
+
+class CardinalityDistribution(HybridMetric):
+    """
+    CARDINALITY_DISTRIBUTION Metric
+
+    Given a column, return the cardinality distribution showing top categories
+    with an "Others" bucket. Only works for concatenable types (strings, enums).
+    """
+
+    threshold_percentage: float = 0.02  # 2% threshold for "Others" bucket
+    min_buckets: int = 5  # Minimum number of top categories to show
+
+    @classmethod
+    def name(cls):
+        return "cardinalityDistribution"
+
+    @property
+    def metric_type(self):
+        return dict
+
+    def fn(
+        self,
+        sample: Optional[DeclarativeMeta],
+        res: Dict[str, Any],
+        session: Optional[Session] = None,
+    ) -> Optional[Dict[str, Any]]:
+        """
+        Build the Cardinality Distribution metric query
+        """
+        if not session:
+            raise AttributeError(
+                "We are missing the session attribute to compute the CardinalityDistribution."
+            )
+
+        if self.col is None:
+            raise AttributeError(
+                "We are missing the column attribute to compute the CardinalityDistribution."
+            )
```

Comment:
nit: can we combine both
```suggestion
        if not session or self.col:
            raise AttributeError(
                f"We are missing the {'session' if not session else 'column'} attribute to compute the CardinalityDistribution."
            )
```